src/metaschema/component-model.txt

```diff
@@ -0,0 +1,148 @@
+# Syntax
+
+assembly-name:as-type (flag:as-type)
+  child-field-or-assembly:as-type
+
+Where:
+
+For old Metaschema:
+- `child-field-or-assembly` is a field if it contains only flags, or an assembly otherwise
+- Cardinality is expressed as follows:
+  - ? (0 or 1) - on a `flag` required="no"; on a field or assembly use singular form.
+  - * (0 to many) - on a field or assembly use plural form.
+  - + (1 to many) - on a field or assembly use plural form with required="yes".
+  - otherwise (1) - no cardinality marker means the flag, field, or assembly is singular and required="yes"
+
+For new Metaschema:
+
+- `child-field-or-assembly` is a field if it contains only flags, or an assembly otherwise
+- Cardinality is expressed as follows:
+  - ? (0 or 1) - this is the default cardinality; no need to specify min-occurs or max-occurs.
+  - * (0 to many) - on a field or assembly use `max-occurs="unbounded"`.
+  - + (1 to many) - on a field or assembly use `min-occurs="1" max-occurs="unbounded"`.
+  - otherwise (1) - no cardinality marker means the flag is `required="yes", and field or assembly is `min-occurs="1"`"
+- The type is specified using the `as-type` marker. If no marker is present, the default `string` type will be used.
+- The marker `v` after the cardinality means that a valid-values list needs to be specified.
+
+# Component Definition
+
+component-definition
+  metadata
+  import-component-definition (href)
```

Comment:
@david-waltermire-nist what does this import denote? Is this a dependent component definition? Or when imported, becomes inline to the component definition being created?